### PR TITLE
55 setup GUI

### PR DESCRIPTION
**Pull request by K-Schubert**
Setup and launch "chatgpt-minimal" frontend service from docker-compose.yml.

See issue https://github.com/CdC-SI/eak-copilot/issues/55.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,62 @@
...
+### Installation Instructions
+
+### Installation
+
+1. **Clone the Repository**
+
+   Begin by cloning the EAK-Copilot repository to your local machine to get the necessary project files.
+
+   ```bash
+   git clone https://github.com/IhrRepository/eak-copilot.git
```
Comment (tabee) [inline README.md:41]: https://github.com/CdC-SI/eak-copilot.git

File: README.md [hunk 2]
```diff
@@ -1,2 +1,62 @@
...
+
+# EAK-Copilot
+
+Welcome to the official repository of the EAK-Copilot challenge, developed as part of the [Innovation Fellowship 2024](https://www.innovationfellowship.ch/). This project is designed to enhance workplace efficiency and foster innovation by providing AI-supported tools that assist employees in their daily tasks.
+
+This repository serves as a proof of concept (PoC), which is slated to conclude in February 2025. However, we are optimistic that the momentum generated by this innovative challenge will attract continued support and development beyond this timeframe.
+
+## Challenge Vision
+
+For a detailed overview of our project's vision and strategic alignment, see our [Vision Document](VISION.md).
```
Comment (tabee) [inline README.md:12]: add comment "comming soon"

File: README.md [hunk 3]
```diff
@@ -1,2 +1,62 @@
...
+    docker-compose build
+    docker-compose up -d
+    ```
+2. **Verifying the Installation**
+   
+    Check the status of the containers to confirm everything is running as expected:
+    ```bash
+    docker-compose ps
+    ```
+    After the containers are successfully started, verify that the application is running correctly by accessing it through your web browser. Navigate to http://localhost:PORT, replacing PORT with the port number specified in your docker-compose.yml.
```
Comment (tabee) [inline README.md:62]: maybe use same port here as we use in env-example file

File: docker-compose.yml [hunk 4]
```diff
@@ -40,30 +44,35 @@ services:
...
       - chatbot
 
   # frontend
-  streamlit:
-    build:
-      context: ./frontend
-      dockerfile: Dockerfile
-    image: copilot-frontend
+  chatgpt-minimal:
+    build: ../chatgpt-minimal
```
Comment (tabee) [inline docker-compose.yml:65]: We are doing it like this at the moment, but we should discuss this point of a separate repository for mvp02. The important thing is that anyone can start a github workspace without forking other repos first. So I think we need a docker that imports our fork by URL path.

File: docker-compose.yml [hunk 5]
```diff
@@ -74,13 +83,12 @@ services:
     ports:
       - "8009:8009"
     volumes:
-      - ./chatbot/app:/app    
-    networks: 
+      - ./chatbot/app:/app
+    networks:
       - copilot-network
     depends_on:
       - db
```
Comment (tabee) [inline docker-compose.yml:90]: chatbot depends_on fastAPI

File: docker-compose.yml (hunk 4 above)
Comment (K-Schubert) [reply docker-compose.yml:65]: Yes, I had some issues with integrating a git clone of our chatgpt-minimal fork in the frontend Dockerfile of the eak-copilot repo. I will create an other issue to deploy the frontend only from the eak-copilot project, and remove the necessity of manually cloning our fork of the chatgpt-minimal repo.

Comment (tabee) [reply docker-compose.yml:65]: @K-Schubert ok. we do this in sprint-02. not important yet. important we have a issue to ckeck this later

File: docker-compose.yml (hunk 5 above)
Comment (K-Schubert) [reply docker-compose.yml:90]: I don't believe the chatbot service requires the fastapi service to be started before as they don't interact at the moment.

File: docker-compose.yml [hunk 6]
```diff
@@ -40,30 +44,35 @@ services:
...
+      - .env
+    environment:
+      - OPENAI_API_KEY=${OPENAI_API_KEY}
+      - REACT_APP_QUERY_AUTOCOMPLETE_API_URL=${REACT_APP_QUERY_AUTOCOMPLETE_API_URL}
+    networks:
       - copilot-network
-    depends_on:
-      - db
-      - chatbot
-      - fastapi
```
[the remark this answers is not in the document]
Comment (K-Schubert) [reply docker-compose.yml:66]: Indeed, I've added the dependency.

File: pgadmin/servers.json
Comment (tabee) [inline pgadmin/servers.json:1]: Ok for now. Please add a new topic.
In pgadmin-container/dockerfile we need to run a bash script that does something like this:

#!/bin/bash
cat << EOF > /pgadmin4/servers.json
{
    "Servers": {
        "1": {
            "Name": "Primary Database",
            "Group": "Servers",
            "Host": "$DATABASE_HOST",
            "Port": 5432,
            "MaintenanceDB": "postgres",
            "Username": "postgres",
            "Password": "$DATABASE_PASSWORD",
            "SSLMode": "prefer"
        }
    }
}
EOF
exec pgadmin4

File: .env.example
Comment (tabee) [inline .env.example:1]: 👍

File: README.md [hunk 7]
```diff
@@ -1,2 +1,62 @@
...
+
+
+# EAK-Copilot
+
+Welcome to the official repository of the EAK-Copilot challenge, developed as part of the [Innovation Fellowship 2024](https://www.innovationfellowship.ch/). This project is designed to enhance workplace efficiency and foster innovation by providing AI-supported tools that assist employees in their daily tasks.
+
+This repository serves as a proof of concept (PoC), which is slated to conclude in February 2025. However, we are optimistic that the momentum generated by this innovative challenge will attract continued support and development beyond this timeframe.
+
+## Challenge Vision
+
```
Comment (tabee) [inline README.md:11]: perfect

File: api/app/main.py [hunk 8]
```diff
@@ -1,21 +1,43 @@
...
+# Load environment variables
+load_dotenv()
+CORS_ALLOWED_ORIGINS = os.environ["CORS_ALLOWED_ORIGINS"]
+POSTGRES_USER = os.environ["POSTGRES_USER"]
+POSTGRES_PASSWORD = os.environ["POSTGRES_PASSWORD"]
+POSTGRES_DB = os.environ["POSTGRES_DB"]
+POSTGRES_HOST = os.environ["POSTGRES_HOST"]
+
+# Setup CORS
+app.add_middleware(
```
Comment (tabee) [inline api/app/main.py:27]: please describe in 1-2 sentences why we are doing this and how tight it is necessary.

**Comment by tabee** [comment]
my code review is finished. i have not tested the code. i assume that it works and wish you a lot of fun with the merge to main.

File: api/app/main.py (hunk 8 above)
Comment (K-Schubert) [reply api/app/main.py:27]: Here we define the service name ("db") of the PostgreSQL database in the docker-compose.yml file. This "db" service runs the PostgreSQL server. I will leave this hardcoded to simplify.

File: pgadmin/servers.json
Comment (K-Schubert) [reply pgadmin/servers.json:1]: I will create a new issue for this.

**Pull request merged**